### ISSUE TITLE
DateTimeField is not deserialized correctly for TriggerChannel

### ISSUE THREAD

**Issue by romank0** [Opus10/django-pgpubsub#27]
When the model with DateTimeField is deserialized the field just get the string value.

This test fails:

```
def test_deserialize_trigger_channel():
    @dataclass
    class MyChannel(TriggerChannel):
        model: Post

    some_datetime = datetime.datetime.utcnow()
    post = Post(content='some-content', date=some_datetime)
    deserialized = MyChannel.deserialize(
        json.dumps(
            {
                'app': 'tests',
                'model': 'Post',
                'old': None,
                'new': {'content': 'some-content', 'date': some_datetime.isoformat()},
            }
        )
    )
    assert deserialized['new'].date == some_datetime
```

This is related to the https://github.com/Opus10/django-pgpubsub/issues/20 as in the root cause of the problem is the same namely the django fields configuration is not taken into account when the row serialized to json in trigger it deserialized back. In this specific case the field type is not taken into account.

**Comment by PaulGilmartin** [comment]
@romank0 Thanks for raising this, this was definitely an oversight when designing the deserialization of trigger based channels. I'm wondering if there's a more general approach than the fix you propose, which works specifically for the case of DateTimeFields (i.e. other fields with particular types will suffer from the same problem of not being deserialized correctly). Perhaps Django's in-built [deserializers](https://docs.djangoproject.com/en/3.2/topics/serialization/#deserializing-data) could be used, but I'm not sure. Do you have any ideas for a more generalised approach? If not, I can still approve and merge your PR for an interim fix.

**Comment by romank0** [comment]
Desearializers do look promising to solve this in general. I'll give them a try some time later. 

My approach to incorporating incomplete solutions (like this fix for particular bug and not the general) is that it is ok do so if two conditions met:
1. the tactical solution does not create problems to create a proper fix later
2. there are tests that cover the tactical fix properly

My reasoning is that this tactical fix is useful (at least for me) today. General fix requires much more effort (specifically handling all edge cases). It does not prevent from proper solution being created and even helps creating it by providing test cases for.

**Comment by PaulGilmartin** [comment]
@romank0 totally agree. I think getting the tactical fix out is definitely better than what we have and it also means there's more time to think about the general one. I'll approve your PR, thanks for contributing!

**Comment by PaulGilmartin** [comment]
@romank0 The fix is now available in v1.0.3 https://pypi.org/project/django-pgpubsub/1.0.3/
